tests/unit/lms/models/application_instance_test.py

```diff
@@ -83,50 +83,22 @@ def test_lms_host_raises_ValueError(self, application_instance, lms_url):
             application_instance.lms_host()
 
     def test_decrypted_developer_secret_returns_the_decrypted_developer_secret(
-        self, application_instance, pyramid_request
-    ):
-        aes_secret = pyramid_request.registry.settings["aes_secret"]
-        application_instance.aes_cipher_iv = Random.new().read(AES.block_size)
-        application_instance.developer_secret = AES.new(
-            aes_secret, AES.MODE_CFB, application_instance.aes_cipher_iv
-        ).encrypt(b"TEST_DEVELOPER_SECRET")
-
-        assert (
-            application_instance.decrypted_developer_secret(aes_secret)
-            == b"TEST_DEVELOPER_SECRET"
-        )
-
-    def test_decryped_developer_secret_returns_None_if_ApplicationInstance_has_no_developer_secret(
-        self, application_instance, pyramid_request
+        self, application_instance, aes_service
     ):
-        aes_secret = pyramid_request.registry.settings["aes_secret"]
+        application_instance.developer_secret = sentinel.developer_secret
+        application_instance.aes_cipher_iv = sentinel.aes_cipher_iv
 
-        assert application_instance.decrypted_developer_secret(aes_secret) is None
+        developer_secret = application_instance.decrypted_developer_secret(aes_service)
 
-    @pytest.mark.parametrize(
-        "developer_secret",
-        [
-            "TEST_DEVELOPER_SECRET",
-            b"TEST_DEVELOPER_SECRET",
-        ],
-    )
-    def test_build_from_lms_url(self, developer_secret, pyramid_request):
-        application_instance = ApplicationInstance.build_from_lms_url(
-            "https://example.com/",
-            "example@example.com",
-            "TEST_DEVELOPER_KEY",
-            developer_secret,
-            pyramid_request.registry.settings["aes_secret"],
-            {},
+        aes_service.decrypt.assert_called_once_with(
+            application_instance.aes_cipher_iv, application_instance.developer_secret
         )
+        assert developer_secret == aes_service.decrypt.return_value
 
-        assert application_instance.consumer_key
-        assert application_instance.shared_secret
-        assert application_instance.lms_url == "https://example.com/"
-        assert application_instance.requesters_email == "example@example.com"
-        assert application_instance.developer_key == "TEST_DEVELOPER_KEY"
-        assert application_instance.developer_secret
-        assert application_instance.settings == {}
+    def test_decryped_developer_secret_returns_None_if_ApplicationInstance_has_no_developer_secret(
+        self, application_instance, aes_service
+    ):
+        assert application_instance.decrypted_developer_secret(aes_service) is None
```

Comment:
It might be nice to set the developer key to `None` here to show the point of the test, even if it's not required.